docs/source/using-discovery/slurm.md

```diff
@@ -0,0 +1,769 @@
+
+:::{code} bash
+scontrol show reservation <reservation_name>
+:::
+
+# Advanced Usage
+Advanced usage of Slurm involves working with multi-node jobs, GPU jobs, and understanding priority and Quality of Service (QoS) parameters. It also involves memory management and the use of environment variables in job scripts.
+
+## Multi-node Jobs
+Multi-node jobs involve executing a single job across multiple nodes. Such jobs are typically used for computationally intensive tasks that require significant parallelization.
+
+### Use-cases: Multi-node Jobs
+Multi-node jobs are used in scenarios where tasks can be broken down into sub-tasks that can be executed in parallel, such as simulation and modeling, machine learning training, or big data analysis.
+
+### Code Example for Multi-node Job Submission
+:::{code} bash
+#!/bin/bash
+#SBATCH -J MultiNodeJob     # Job name
+#SBATCH -N 4                # Number of nodes
+#SBATCH -n 16               # Number of tasks
+
+# Your program/command here
+srun ./my_program
+:::
+
+## GPU Jobs
+Slurm can also manage GPU resources, allowing you to specify GPU requirements in your job scripts.
+
+### Use-cases: GPUs
+GPU jobs are used in scenarios where tasks are parallelized and can benefit from the high computational capabilities of GPUs, such as machine learning and deep learning workloads, image processing, or simulations.
+
+### Code Example for GPU Job Submission
+:::{code} bash
+#!/bin/bash
+#SBATCH -J GPUJob            # Job name
+#SBATCH -N 1                 # Number of nodes
+#SBATCH -n 1                 # Number of tasks
+#SBATCH --gres=gpu:2         # Number of GPUs
+
+# Your program/command here
+srun ./my_gpu_program
+:::
+
+## Priority and QoS
+Slurm uses priority and Quality of Service (QoS) parameters to determine the order in which jobs are scheduled.
+
+### Understanding Job Priorities and Quality of Service (QoS) Parameters
+The job priority is a numerical value assigned to each job, determining its position in the queue. Higher priority jobs are scheduled before lower priority ones.
+Quality of Service (QoS) parameters control various job limits, such as the maximum allowed job runtime, the maximum number of CPUs or nodes a job can use, etc.
+
+### Code Example to Manipulate Job Priority
+:::{code} bash
+scontrol update jobid=<job_id> priority=<new_priority>
+:::
+
+## Memory Management
+In Slurm, memory allocation can be controlled on the job or task level using the `--mem` or `--mem-per-cpu` options, respectively.
+
+### Code example for specifying memory in job scripts
+:::{code} bash
+#!/bin/bash
+#SBATCH -J MyJob             # Job name
+#SBATCH -N 1                 # Number of nodes
+#SBATCH -n 4                 # Number of tasks
+#SBATCH --mem=8G             # Memory for the entire job
+#SBATCH --mem-per-cpu=2G     # Memory per task (CPU)
```

Comment:
You might want to break this up to two examples because you cannot specify both of these in a single script.